backend/api/src/realms.rs

```diff
@@ -29,10 +32,31 @@ impl Realm {
         Id::realm(self.parent_key)
     }
 
+    fn path(&self) -> &str {
+        &self.path
+    }
+
     fn parent(&self, context: &Context) -> &Realm {
         &context.realm_tree.realms[&self.parent_key]
     }
 
+    fn parents(&self, context: &Context) -> Vec<&Realm> {
+        let mut parents = VecDeque::new();
+        let mut generation = self;
+
+        // Add all the parents up to the root (which might be ourselves already,
+        // in which case this never runs)
+        while generation.parent_key != 0 {
+            generation = context.realm_tree.realms.get(&generation.parent_key).unwrap();
+            parents.push_front(generation);
+        }
+        // Always add the root
+        generation = context.realm_tree.realms.get(&generation.parent_key).unwrap();
+        parents.push_front(generation);
```

Comment:
```suggestion
        while generation.key != 0 {
            generation = context.realm_tree.realms.get(&generation.parent_key).unwrap();
            parents.push_front(generation);
        }
```

Shouldn't this work equally well?

Reply:
Maybe, but it's done completely differently anyway now, so ... :sweat_smile: